tests_pycsob/test_api.py

```diff
@@ -21,6 +22,121 @@
 responses = Responses(package='requests.packages.urllib3')
 
 
+class CartItemTests(TestCase):
+
+    def test_to_dict(self):
+        cart_item = CartItem(name="01234567890123456789xxx", quantity=5, amount=1000)
+        expected = OrderedDict([
+            ("name", "01234567890123456789"),  # trimmed to 20 chars
+            ("quantity", 5),
+            ("amount", 1000),
+        ])
+        self.assertEqual(cart_item.to_dict(), expected)
+
+    def test_to_dict_description(self):
+        cart_item = CartItem(name="test", quantity=5, amount=1000, description="x" * 50)
+        expected = OrderedDict([
+            ("name", "test"),
+            ("quantity", 5),
+            ("amount", 1000),
+            ("description", "x" * 40),  # trimmed to 40 chars
+        ])
+        self.assertEqual(cart_item.to_dict(), expected)
+
+
+class CustomerDataTests(TestCase):
+
+    def test_to_dict_minimum(self):
+        customer_data = CustomerData(name="test " * 10)
+        expected = OrderedDict([
+            ("name", "test " * 9),  # trimmed to 45 chars
+        ])
+        self.assertEqual(customer_data.to_dict(), expected)
+
+    def test_to_dict_complex(self):
+        customer_data = CustomerData(
+            name="test",
+            mobile_phone="+420.123456789",
+            account = CustomerAccount(created_at="2024-06-25T13:30:15+02:00", changed_at="2024-06-25T15:21:07+02:00"),
+            login = CustomerLogin(auth=CustomerLoginType.ACCOUNT, auth_at="2024-06-26T08:53:47+02:00"),
```

Comment:
This shouldn't go through until `__str__` is there or `.value` is used.

Reply:
But it seems to work, we have `_format_auth` method to handle it.